pkg/resiliency/policy.go

```diff
@@ -26,63 +26,68 @@ import (
 
 type (
 	// Operation represents a function to invoke with resiliency policies applied.
-	Operation func(ctx context.Context) error
+	Operation func(ctx context.Context) (any, error)
 
 	// Runner represents a function to invoke `oper` with resiliency policies applied.
-	Runner func(oper Operation) error
+	Runner func(oper Operation) (any, error)
 )
 
-// Policy returns a policy runner that encapsulates the configured
-// resiliency policies in a simple execution wrapper.
+type doneCh struct {
+	res any
+	err error
+}
+
+// Policy returns a policy runner that encapsulates the configured resiliency policies in a simple execution wrapper.
 func Policy(ctx context.Context, log logger.Logger, operationName string, t time.Duration, r *retry.Config, cb *breaker.CircuitBreaker) Runner {
-	return func(oper Operation) error {
+	return func(oper Operation) (any, error) {
 		operation := oper
 		if t > 0 {
-			// Handle timeout.
-			// TODO: This should ideally be handled by the underlying service/component. Revisit once those understand contexts.
+			// Handle timeout
+			// TODO: This should ideally be handled by the underlying service/component. Revisit once those understand contexts
 			operCopy := operation
-			operation = func(ctx context.Context) error {
+			operation = func(ctx context.Context) (any, error) {
 				ctx, cancel := context.WithTimeout(ctx, t)
 				defer cancel()
 
-				done := make(chan error)
+				done := make(chan doneCh)
 				go func() {
-					done <- operCopy(ctx)
+					rRes, rErr := operCopy(ctx)
+					done <- doneCh{rRes, rErr}
 				}()
 
 				select {
-				case err := <-done:
-					return err
+				case v := <-done:
```

Comment:
Do we want to timeout here? (Or is that what the comment on 45 is alluding to?)

Reply:
The timeout happens with the context. Line 61 below is what executes the timeout